api/api/files/routes.go

```diff
@@ -358,15 +381,18 @@ func GetFileList(c *gin.Context) {
 	}
 
 	ctx := context.Background()
-	dbClient := firestore.NewClient(ctx)
+	dbClient, err := firestore.Service.NewClient(ctx)
+	if err != nil {
+		log.Panicln(err)
+	}
 	defer dbClient.Close() // nolint: errcheck
 
-	docs, err := firestore.ListByTags(ctx, dbClient, tags, orderNo, size)
+	docs, err := dbClient.ListByTags(ctx, tags, orderNo, size)
 	if err != nil {
 		log.Panicln(err)
 	}
 
-	results := []FileMeta{}
+	var results = []FileMeta{} // An empty slice is intended for the JSON response instead of nil.
```

Comment:
For this variable, we use `var results = []FileMeta{}` intentionally so that the default response to the frontend would be an empty list instead of nil. However the implementation can be changed if using `var results []FileMeta` makes more sense : )

Reply:
It would still be an empty slice if you declare it as I've asked. See:
https://go.dev/play/p/TKMEPVKOXSZ

Why wouldn't you want to check for nil?